Streetcode/Streetcode.BLL/Middleware/ApiRequestResponseMiddleware.cs

```diff
@@ -0,0 +1,127 @@
+using Streetcode.BLL.Interfaces.Logging;
+using Streetcode.BLL.Services.Logging;
+
+namespace Streetcode.BLL.Middleware
+{
+    public class ApiRequestResponseMiddleware : IMiddleware
+    {
+        private readonly ILoggerService _loggerService;
+        private string[] _loggerPropertiesExceptions;
+
+        public ApiRequestResponseMiddleware(ILoggerService loggerService, IConfiguration configuration)
+        {
+            _loggerService = loggerService;
+            _loggerPropertiesExceptions = configuration.GetSection("Serilog:PropertiesToIgnore").Get<string[]>();
+        }
+
+        public async Task InvokeAsync(HttpContext context, RequestDelegate next)
+        {
+            var originalBodyStream = context.Response.Body;
+
+            var request = await GetRequestAsTextAsync(context.Request);
+
+            _loggerService.LogInformation(request);
+
+            await using var responseBody = new MemoryStream();
+            context.Response.Body = responseBody;
+
+            await next(context);
+
+            var response = await GetResponseAsTextAsync(context.Response);
+
+            var filteredResponse = GetFilteredResponse(response);
+
+            _loggerService.LogInformation(filteredResponse);
+
+            await responseBody.CopyToAsync(originalBodyStream);
+        }
+
+        private async Task<string> GetRequestAsTextAsync(HttpRequest request)
+        {
+            var requestBody = request.Body;
+
+            request.EnableBuffering();
+
+            var buffer = new byte[Convert.ToInt32(request.ContentLength)];
+
+            await request.Body.ReadAsync(buffer, 0, buffer.Length);
+
+            var bodyAsText = Encoding.UTF8.GetString(buffer);
+
+            request.Body = requestBody;
+
+            return $"{request.Scheme} {request.Host} {request.Path} {request.QueryString} {bodyAsText}";
+        }
+
+        private async Task<string> GetResponseAsTextAsync(HttpResponse response)
+        {
+            response.Body.Seek(0, SeekOrigin.Begin);
+
+            var encoding = Encoding.UTF8;
+
+            var text = await new StreamReader(response.Body, encoding).ReadToEndAsync();
+
+            response.Body.Seek(0, SeekOrigin.Begin);
+
+            return text;
+        }
+
+        private string GetFilteredResponse(string response)
+        {
+            try
+            {
+                var jsonObject = JToken.Parse(response);
+
+                if (jsonObject.Type == JTokenType.Array)
+                {
+                    var jsonArray = (JArray)jsonObject;
+
+                    foreach (var item in jsonArray)
+                    {
+                        TruncateProperties(item);
+                    }
+                }
+                else if (jsonObject.Type == JTokenType.Object)
+                {
+                    TruncateProperties(jsonObject);
+                }
+
+                return jsonObject.ToString();
+            }
+            catch (Exception ex)
+            {
```

Comment:
In case the exception all data returns raw, so if I will return response, I will log into console all uncutted base64 (very long file), but my task was to get rid of this, so I return null, but it also can be anything except response.